test/test_string_array.cpp

```diff
@@ -130,3 +130,73 @@ TEST(test_string_array, string_array_cmp) {
   ret = rcutils_string_array_fini(&incomplete_string_array);
   ASSERT_EQ(RCUTILS_RET_OK, ret);
 }
+
+TEST(test_string_array, string_array_resize) {
+  auto allocator = rcutils_get_default_allocator();
+  auto failing_allocator = get_failing_allocator();
+  auto invalid_allocator = rcutils_get_zero_initialized_allocator();
+  rcutils_ret_t ret;
+
+  ret = rcutils_string_array_resize(nullptr, 8);
```

Comment:
You may want to call rcutils_reset_error(); when a error happens otherwise you will show some rcl overwritten errors in the terminal.

Reply:
Done in 122afc1.